chrome/content/zotero/itemPane.js

```diff
@@ -185,6 +189,132 @@ var ZoteroItemPane = new function() {
 	}
 	
 	
+	this.translateSelectedItems = Zotero.Promise.coroutine(function* () {
+		if (_blockTranslate) {
+			_blockTranslate = false;
+			return;
+		}
+		var collectionID = _translationTarget.objectType == 'collection' ? _translationTarget.id : undefined;
+		var items = ZoteroPane_Local.itemsView.getSelectedItems();
+		if (!_translationTarget) {
+			_translationTarget = Zotero.Libraries.userLibrary;
+		}
+		for (let item of items) {
+			yield item.translate(_translationTarget.libraryID, collectionID);
+		}
+	});
+	
+	
+	this.buildTranslateSelectContextMenu = function (event) {
+		_blockTranslate = true;
+		var menu = document.getElementById('zotero-item-addTo-menu');
+		if (event.target != menu) {
+			return;
+		}
+		// Clear previous items
+		while (menu.firstChild) {
+			menu.removeChild(menu.firstChild);
+		}
+		
+		var libraries = Zotero.Libraries.getAll();
+		for (let library of libraries) {
+			if (!library.editable || library.libraryType == 'publications') {
+				continue;
+			}
+			let libraryMenu = _createTranslationMenuForTarget(library);
+			menu.appendChild(libraryMenu);
+		}
+	};
```

Comment:
We'll want to do the same thing in zoteroPane.js for the items list context menu, and possibly other places, so some of this should just be a function in utilities_internal.js (along with `createTranslationMenuForTarget`) that takes a menu element to populate and an array of libraries.